mitmproxy/certs.py

```diff
@@ -106,15 +106,24 @@ def dummy_cert(privkey, cacert, commonname, sans, organization):
     cert.gmtime_adj_notBefore(-3600 * 48)
     cert.gmtime_adj_notAfter(DEFAULT_EXP_DUMMY_CERT)
     cert.set_issuer(cacert.get_subject())
-    if commonname is not None and len(commonname) < 64:
+    is_valid_commonname = (
+        commonname is not None and len(commonname) < 64
```

Comment:
`<= 64`? I just picked up what you said in #3981 ("the CN field is limited to 64 characters") but maybe there's something I don't know where the 64th character is needed (trailing dot or whatever?)

Also this sounds like something that could be beautifully unit tested. Sorry for bugging you :grin:

Reply:
I've just copied what's there. Let's maybe be extra safe here and keep it as-is, it shouldn't hurt after all. And yes, this should get a test. If someone wants to complete this please feel free, otherwise this will sit a couple of weeks until I have a bit of spare time :)